### PR TITLE
BREAKING: Bump to Node 16

### PR DESCRIPTION
**Pull request by FrederikBolding**
## Description

Bumps the minimum Node version of the packages to `16.18.24`.

## Changes

- **BREAKING**: Bumps the minimum Node version of all packages to `16.18.24`

### CODE REVIEW

**Comment by FrederikBolding** [comment]
ty for the updates! I swear I looked for docs that mentioned Node 14 but didn't find any 😅

**Comment by FrederikBolding** [comment]
> ...However, should we make sure each package has a non-major release published before merging this @FrederikBolding ? Considering this repo does not have as straightforward a flow for backports, we might as well get any pending non-breaking changes published first.
> 
> WDYT?

Both the extension and mobile are to my knowledge ready for Node 16. So not sure if there is a reason to wait? 🤔

**Comment by FrederikBolding** [comment]
> I don't have a scenario in mind - I just view is as standard practice and you never know when The Unknown Unknown rears a head.
> 
> And in the off-chance we have external dependents of these packages with different runtime constraints, this may be a big change for such users.

Fair enough, I am merely trying to unblock https://github.com/MetaMask/core/pull/1184 and thought it was easier to do a release after all had been merged. But I'm also fine with us publishing a release first. Doesn't look like many packages have outstanding changes though

File: docs/contributing.md
Comment (Gudahtt) [inline docs/contributing.md:17]: Nit: Seems that a newline was accidentally removed here

```suggestion
> **Note**
> `workspaceName` in these commands is the `name` field within a package's `package.json`, e.g., `@metamask/address-book-controller`, not the directory where it is located, e.g., `packages/address-book-controller`.
```

**Comment by Gudahtt** [comment]
I have applied the `DO-NOT-MERGE` label for now, until the next release is published. I'll ensure we get it published tomorrow.

**Comment by FrederikBolding** [comment]
This should be good to review now!

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Comment by Gudahtt** [comment]
Looks like we need to run `lint:fix`

**Comment by FrederikBolding** [comment]
@Gudahtt That is what resulted in the contributing.md change that you requested reverted 😅

File: docs/contributing.md
Comment (FrederikBolding) [reply docs/contributing.md]: Yeah, not sure what is going on here. This is what Prettier wants

File: docs/contributing.md
Comment (Gudahtt) [inline docs/contributing.md:17]: This seems to work:

```suggestion
> **Note**
>
```

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**
